policy/policy_handler/handler.go

```diff
@@ -0,0 +1,223 @@
+package policy_handler
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/atomist-skills/go-skill"
+	"github.com/atomist-skills/go-skill/policy/data"
+	"github.com/atomist-skills/go-skill/policy/goals"
+	"github.com/atomist-skills/go-skill/policy/storage"
+	"github.com/atomist-skills/go-skill/util"
+	"olympos.io/encoding/edn"
+)
+
+type (
+	EvaluatorSelector func(ctx context.Context, req skill.RequestContext, goal goals.Goal, dataSource data.DataSource) (goals.GoalEvaluator, error)
+
+	Handler interface {
+		Start()
+	}
+
+	subscriptionProvider func(ctx context.Context, req skill.RequestContext) ([][]edn.RawMessage, skill.Configuration, error)
+	dataSourceProvider   func(ctx context.Context, req skill.RequestContext) ([]data.DataSource, error)
+	transactionFilter    func(ctx context.Context, req skill.RequestContext) bool
+
+	EventHandler struct {
+		// parameters
+		evalSelector      EvaluatorSelector
+		subscriptionNames []string
+
+		// hooks used by opts
+		subscriptionDataProviders []subscriptionProvider
+		dataSourceProviders       []dataSourceProvider
+		transactFilters           []transactionFilter
+	}
+
+	Opt func(handler *EventHandler)
+)
+
+func NewPolicyEventHandler(subscriptionNames []string, evalSelector EvaluatorSelector, opts ...Opt) EventHandler {
+	p := EventHandler{
+		subscriptionNames: subscriptionNames,
+		evalSelector:      evalSelector,
+	}
+
+	for _, o := range opts {
+		o(&p)
+	}
+
+	return p
+}
+
+func (h EventHandler) Start() {
+	handlers := skill.Handlers{}
+	for _, n := range h.subscriptionNames {
+		handlers[n] = h.handle
+	}
+
+	skill.Start(handlers)
+}
+
+// EvaluateGoals runs the goal evaluation and returns the results after transacting them.
```

Comment:
Out of date comment?

Reply:
7a40d3a58fe77f9be38ae2dc28db933167030ec6